lib/rspec/core/formatters/exception_presenter.rb

```diff
@@ -31,8 +31,44 @@ def colorized_message_lines(colorizer=::RSpec::Core::Formatters::ConsoleCodes)
           end
         end
 
-        def formatted_backtrace
-          backtrace_formatter.format_backtrace(exception_backtrace, example.metadata)
+        def formatted_backtrace(exception=@exception)
+          lines = backtrace_formatter.format_backtrace((exception.backtrace || []), example.metadata)
+          potential_lines = formatted_cause(exception)
+
+          lines += potential_lines unless potential_lines.empty?
+
+          lines
+        end
+
+        if RSpec::Support::RubyFeatures.supports_exception_cause?
+          def formatted_cause(exception)
+            exceptions = []
+
+            while exception.respond_to?(:cause) && exception.cause
```

Comment:
A few of the tests within `describe '#fully_formatted'` in https://github.com/rspec/rspec-core/blob/master/spec/rspec/core/formatters/exception_presenter_spec.rb fail because `instance_double` received an unexpected message `:cause`. So in order to side step this issue and keep the diff small I decided to use `exception.respond_to?(:cause)`. That way I would not need to add a `before` block stubbing out the method or stub it out in each particular instance. 

However now that I've finished the initial implementation and this pr has been reviewed I think it's good that I stub this method out in each `instance_double`, what would you do in this case?